File: app/src/main/java/cc/smartcash/wallet/Models/SmartTextRoot.java
package cc.smartcash.wallet.Models;

import android.os.Parcel;
import android.os.Parcelable;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import java.io.Serializable;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
        "status",
        "data"
})
@JsonIgnoreProperties(ignoreUnknown = true)
public class SmartTextRoot implements Serializable, Parcelable {

    public final static Parcelable.Creator<SmartTextRoot> CREATOR = new Creator<SmartTextRoot>() {


        @SuppressWarnings({
                "unchecked"
        })
        public SmartTextRoot createFromParcel(Parcel in) {
            return new SmartTextRoot(in);
        }

        public SmartTextRoot[] newArray(int size) {
            return (new SmartTextRoot[size]);
        }

    };
    private final static long serialVersionUID = -3486050116655913572L;
    @JsonProperty("status")
    private String status;
    @JsonProperty("data")
    private SmartTextData data;

    protected SmartTextRoot(Parcel in) {
        this.status = ((String) in.readValue((String.class.getClassLoader())));
        this.data = ((SmartTextData) in.readValue((SmartTextData.class.getClassLoader())));
    }

    public SmartTextRoot() {
    }

    @JsonProperty("status")
    public String getStatus() {
        return status;
    }

    @JsonProperty("status")
    public void setStatus(String status) {
        this.status = status;
    }

    @JsonProperty("data")
    public SmartTextData getData() {
        return data;
    }

    @JsonProperty("data")
    public void setData(SmartTextData data) {
        this.data = data;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(status);
        dest.writeValue(data);
    }

    public int describeContents() {
        return 0;
    }

}

File: app/src/main/java/cc/smartcash/wallet/Models/SmartTextRoot.kt
package cc.smartcash.wallet.Models

import android.os.Parcel
import android.os.Parcelable

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonPropertyOrder

import java.io.Serializable

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder("status", "data")
@JsonIgnoreProperties(ignoreUnknown = true)
class SmartTextRoot : Serializable, Parcelable {
    @JsonProperty("status")
    @get:JsonProperty("status")
    @set:JsonProperty("status")
    var status: String? = null
    @JsonProperty("data")
    @get:JsonProperty("data")
    @set:JsonProperty("data")
    var data: SmartTextData? = null

    private constructor(`in`: Parcel) {
        this.status = `in`.readValue(String::class.java.classLoader) as String
        this.data = `in`.readValue(SmartTextData::class.java.classLoader) as SmartTextData
    }

    constructor()

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(status)
        dest.writeValue(data)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<SmartTextRoot> {
        override fun createFromParcel(parcel: Parcel): SmartTextRoot {
            return SmartTextRoot(parcel)
        }

        override fun newArray(size: Int): Array<SmartTextRoot?> {
            return arrayOfNulls(size)
        }
    }


}